src/service/video-manager-impl.js

```diff
@@ -436,6 +540,136 @@ class VideoEntry {
   }
 
   /**
+   * Called when the video's position in the viewport changed.
+   * @param {PositionInViewportEntryDef} newPos
+   * @private
+   */
+  onPositionChanged_(newPos) {
+
```

Comment:
we wanna short circuit this function as much as possible. few conditions that I can think of where we wanna return immediately

-this.isLoaded_ is false
-is not manually playing and is not docked.